package/setup.py

```diff
@@ -0,0 +1,170 @@
+    fname = re.findall('filename=(.+)', cd)
+    if len(fname) == 0:
+        return None
+    return fname[0]
+
+def get_latest_release(installation_path):
+    import requests
+
+    url = 'https://api.github.com/repos/MetaCell/PsyNeuLinkView/releases'
+    headers = {'Accept': 'application/vnd.github+json','Authorization': 'Bearer JWT', 'X-GitHub-Api-Version' : '2022-11-28'}
+    r = requests.get(url, allow_redirects=True)
+    releases = json.loads(r.text)
+    assets = releases[0]["assets"]
+
+    target_release = None
+    for asset in assets :
+        if platform.system().lower() in asset['name'] :
+            target_release = asset["browser_download_url"]
+
+    logging.info("System detected %s :", platform.system())
+    logging.info("Target release url found %s :", target_release)
+    logging.info("Downloading release to %s...", installation_path)
+    release_download = requests.get(target_release, allow_redirects=True)
+
+    filename = get_filename_from_cd(release_download.headers.get('content-disposition'))
+    tar_location = os.path.join(installation_path, filename)
+    logging.info("Writing release to %s...", tar_location)
+    open(tar_location, 'wb').write(release_download.content)
+
+    logging.info("Opening compressed file %s", filename)
+    tar = tarfile.open(tar_location)
+    extract_location = "/usr/local/bin"
+    tar.extractall(path=extract_location)
+    tar.close()
+    logging.info("Release file uncompressed at : %s", extract_location)
+
+    application = os.path.join(extract_location, "psyneulinkviewer-linux-x64/psyneulinkviewer")
+    symlink = "/usr/local/bin/psyneulinkviewer"
+    logging.info("Creating symlink at : %s", symlink)
+    logging.info("Application at : %s", application)
+    try:
+       if os.path.islink(symlink):
+           os.remove(symlink)
+       os.symlink(application, symlink)
+    except OSError as e:
+       logging.error("Error applying symlin %f ", e)
+
+    logging.info("Symlink created")
+
+    logging.info("*** To launch the application run : **** ")
+    logging.info(" %s " ,symlink)
+
+def prerequisites():
+    check_os()
+    check_python()
+    check_conda()
+    check_rosetta()
+    check_graphviz()
+    check_psyneulink()
+    get_latest_release(os.path.dirname(os.path.realpath(__file__)))
+
+class InstallCommand(install):
+    user_options = install.user_options + [
+        ('path=', None, 'an option that takes a value')
+    ]
+
+    def initialize_options(self):
+        install.initialize_options(self)
+        self.path = None
+
+    def finalize_options(self):
+        # Validate options
+        if self.path is None:
+            self.path = os.path.dirname(os.path.realpath(__file__))
+        super().finalize_options()
+
+
+    def run(self):
+        global path
+        path = self.path # will be 1 or None
+        install.run(self)
+
+setup(
+    name="psyneulinkview",
+    version="0.0.1",
+    setup_requires=['requests'],
+    entry_points={
```

Comment:
from the setup tool documentation:

```
Distributions can specify console_scripts entry points, each referring to a function. When pip (or another console_scripts aware installer) installs the distribution, it will create a command-line wrapper for each entry point.
```

infact, when I check the pip installation I can see

```
(base) ddelpiano@usop:~$ which psyneulinkviewer 
/home/ddelpiano/miniconda3/bin/psyneulinkviewer
````

This could cause some issues, since we are also linking the psyneulinkviewer executable extracted in the /usr/local/bin folder. Depending on which folder comes first in the user PATH, one or the other will never be executed.

I would recommend that we change the entry_points to the cmdclass so that we can perform the installation DURING the pip install and NOT AFTER the pip install, some info here [link](https://niteo.co/blog/setuptools-run-custom-code-in-setup-py).

Reply:
Done